components/builder-depot-client/src/lib.rs

```diff
@@ -398,6 +398,50 @@ impl Client {
         )
     }
 
+    /// Create secret for an origin
+    ///
+    /// # Failures
+    ///
+    /// * Remote Builder is not available
+    pub fn create_origin_secret(&self, origin: &str, token: &str, key: &str, secret: &str) -> Result<()> {
```

Comment:
How does one update a secret? Presumably by calling this function again? I'm wondering if there's utility in having there be any distinction in the interface about whether there was already a value for the provided `key` (such as a `Result<bool>` return) or requiring something like an `overwrite: bool` parameter. I don't actually know; just throwing that idea out there.

Reply:
right now there will be a unique constraint on the key name per origin. Users will have to delete and recreate a key if the want to reuse it. This behavior should be consistent with Travis CI

Reply:
That sounds good. Where is the the `depot/origins/{}/secret` implementation? Also, might be worth a note in the help.

Reply:
https://github.com/habitat-sh/builder/pull/232